<a href="https://colab.research.google.com/github/notyourfriends/Playground/blob/main/Tech_Layoffs.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
import pandas as pd

In [ ]:
data = pd.read_csv('/content/drive/MyDrive/CSVs/tech_layoffs.csv')

In [ ]:
data.columns

Index(['company', 'total_layoffs', 'impacted_workforce_percentage',
       'reported_date', 'industry', 'headquarter_location', 'sources',
       'status', 'additional_notes'],
      dtype='object')

In [ ]:
data.head(5)

company total_layoffs impacted_workforce_percentage reported_date  \
0            C2FO            20                             2     12/9/2022   
1        VideoAmp       Unclear                             2     9/19/2022   
2        Amperity            13                             3     8/20/2022   
3         Addepar            20                             3     1/18/2023   
4  Thirty Madison            24                             3     5/13/2022   

                  industry headquarter_location                       sources  \
0        Fintech, payments          Leawood, KS  Kansas City Business Journal   
1    Advertising platforms          Los Angeles                  Company memo   
2          Saas, analytics              Seattle                      Geekwire   
3  fintech, data analytics    Mountain View, CA                       Fortune   
4              Health Care             New York                         Axios   

    status                                   additional_notes  
0  Private                                                NaN  
1  Private                                                NaN  
2  Private                                                NaN  
3  Private                                                NaN  
4  Private  Per Axios, 22 of the employees laid off were f...

In [21]:
dn = data['sources'].value_counts()
dn

TechCrunch          52
The Information     23
Business Insider    23
Calcalistech        20
LinkedIn            16
                    ..
CityWire             1
OneTrust             1
Mobi Health News     1
Homeward             1
Doma                 1
Name: sources, Length: 221, dtype: int64

based on sources, most of the data acquired from TechCrunch

In [27]:
industry_counts = data.groupby('industry').size().reset_index(name='count').sort_values(by='count', ascending=False)
industry_counts

industry  count
91              Fintech     24
109         Health Care     17
166            PropTech     15
63           E-commerce     13
52        Cybersecurity     10
..                  ...    ...
121  IT, future of work      1
122        IT, software      1
125  Insurance, fintech      1
126           Insurtech      1
288         video games      1

[289 rows x 2 columns]

ternyata ada golongan paling banyak layoff ada di industri Fintech

berhubung namanya tidak seragam dan ada simbol, saatnya menyelaraskan nama tersebut

In [29]:
ind = []
for i in data['industry'].to_list():
    ind.append(i.split(' ')[-1])

ind = [x.lower() for x in ind]
ind = [x.replace("-", "") for x in ind]

data['industry'] = ind

In [30]:
data['industry'].head(10)

0         payments
1        platforms
2        analytics
3        analytics
4             care
5             tech
6    cybersecurity
7        ecommerce
8         wellness
9             tech
Name: industry, dtype: object

In [32]:
industry_counts = data.groupby('industry').size().reset_index(name='count').sort_values(by='count', ascending=False)
industry_counts

industry  count
53      fintech     35
107    software     32
94     proptech     30
40    ecommerce     29
20         care     27
..          ...    ...
49       events      1
46   esignature      1
44       energy      1
42      editing      1
128        work      1

[129 rows x 2 columns]

In [33]:
data['industry'].to_list()

['payments',
 'platforms',
 'analytics',
 'analytics',
 'care',
 'tech',
 'cybersecurity',
 'ecommerce',
 'wellness',
 'tech',
 'software',
 'crypto',
 'care',
 'transportation',
 'cybersecurity',
 'ecommerce',
 'logistics',
 'ecommerce',
 'tech',
 'ecommerce',
 'care',
 'cryptocurrency',
 'ecommerce',
 'productivity',
 'transportation',
 'ecommerce',
 'insurtech',
 'games',
 'fintech',
 'crypto',
 'retail',
 'logistics',
 'intelligence',
 'logistics',
 'tools',
 'care',
 'ecommerce',
 'fintech',
 'analytics',
 'robotics',
 'finance',
 'cannabis',
 'logistics',
 'ar',
 'media/entertainment',
 'media/entertainment',
 'tech',
 'journalism',
 'media',
 'cybersecurity',
 'analytics',
 'software',
 'software',
 'finance',
 'saas',
 'learning',
 'enterprise',
 'fintech',
 'tech',
 'security',
 'transportation',
 'processing',
 'cryptocurrency',
 'retail',
 'blockchain',
 'proptech',
 'marketing',
 'agtech',
 'care',
 'crypto',
 'proptech',
 'fintech',
 'fintech',
 'fintech',
 'fintech',
 'ec

semua data udah seragam sekarang

# Preprocess

In [ ]:
data.isnull().sum()

company                            0
total_layoffs                      0
impacted_workforce_percentage      0
reported_date                      0
industry                           0
headquarter_location               0
sources                            0
status                             0
additional_notes                 467
dtype: int64

ada sebanyak 467 data kosong di "additional_notes", drop aja ga penting juga wkwk

In [ ]:
data.drop('additional_notes', inplace=True, axis=1)

In [ ]:
data.head()

In [ ]:
data.columns

Index(['company', 'total_layoffs', 'impacted_workforce_percentage',
       'reported_date', 'industry', 'headquarter_location', 'sources',
       'status'],
      dtype='object')

# EDA